sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkMeter.java

```diff
@@ -10,15 +10,38 @@
 import io.opentelemetry.api.metrics.LongCounterBuilder;
 import io.opentelemetry.api.metrics.LongUpDownCounterBuilder;
 import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.api.metrics.MeterProvider;
 import io.opentelemetry.sdk.common.InstrumentationScopeInfo;
 import io.opentelemetry.sdk.metrics.data.MetricData;
 import io.opentelemetry.sdk.metrics.internal.export.CollectionInfo;
 import io.opentelemetry.sdk.metrics.internal.state.MeterProviderSharedState;
 import io.opentelemetry.sdk.metrics.internal.state.MeterSharedState;
 import java.util.Collection;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
 
 /** {@link SdkMeter} is SDK implementation of {@link Meter}. */
 final class SdkMeter implements Meter {
+
+  /**
+   * Instrument names MUST conform to the following syntax.
+   *
+   * <ul>
+   *   <li>They are not null or empty strings.
+   *   <li>They are case-insensitive, ASCII strings.
+   *   <li>The first character must be an alphabetic character.
+   *   <li>Subsequent characters must belong to the alphanumeric characters, '_', '.', and '-'.
+   *   <li>They can have a maximum length of 63 characters.
+   * </ul>
+   */
+  private static final String VALID_INSTRUMENT_PATTERN =
```

Comment:
Nit that it's a bit mysterious for _PATTERN to not be the Pattern. I think this is split out mainly to use in the log message but I feel this regex-fu will not be all that helpful in a log message and would probably write prose instead, maybe "must be a non-empty string up to 63 characters including alphanumeric, _, ., -, and starting with a letter". Or something like that

Reply:
(Also, it this is about a log message you could simply use `Pattern#pattern()` or `Pattern#toString()`)